sanity-io/sanity
# fix(core): non valid file types for image and file inputs will be greyed out

### Description

In the past when using the default asset source, you could pick any of the files / images there regardless of whether the input had `accept` limitations (type limitations)

You can still see their usage and delete them, just not upload them to the input.

| Image (png) | file (pdfs) |
|-------|-------|
|  <img width="777" alt="image" src="https://user-images.githubusercontent.com/6951139/217217531-da9d8e42-c1f3-4943-9de4-97961a9076be.png"> | <img width="777" alt="image" src="https://user-images.githubusercontent.com/6951139/217217631-4fb27712-b65a-42cf-95b5-8f427da4bd8a.png">  |

Fix issue where the file type was "overflowing" into the next table cell in the file input
<img width="662" alt="Pasted Graphic" src="https://user-images.githubusercontent.com/6951139/218752547-d66a733d-1597-4b7b-abbc-b29862ac7a77.png">


### What to review

#### Image - Without `accept`
1. Go to test-studio
2. Standard inputs -> image -> Main Image field
4. Press select -> uploaded images
5. You should be able to select and upload any item

#### Image - With `accept`
1. Go to test-studio
2. Standard inputs -> image -> PNG image field
3. Press select -> uploaded images
4. Check that all assets are specifically pngs (you can check by trying to delete them and seeing the extension, for example)

#### File - Without `accept`
1. Go to test-studio
2. Standard inputs -> file -> Simple file (field)
4. Press browse
5. You should be able to select and upload any item

#### File - With `accept`
1. Go to test-studio
2. Standard inputs -> file -> File only accepts pdf field
3. Press browse
4. Check that all assets are pdfs

#### File - With multiple `accept` (png, jpg)
1. Go to test-studio
2. Standard inputs -> image -> JPEG & PNG image
4. Press browse
5. Check that all assets are either png or jpg (you can check by trying to delete them and seeing the extension, for example)
 
Check that it looks good on both *light and dark mode*

### Notes for release

While browsing, the File and Image input will only allow you to select items that are of a valid type
